### PR TITLE
Reduce oscillation when destination is blocked

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Tune some factors to reduce oscillation when destination is blocked

### Testing Done
grsim

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.h [hunk 1]
```diff
@@ -192,8 +192,13 @@ class ThetaStarPathPlanner : public PathPlanner
...
 
-    static constexpr double CLOSE_TO_DEST_THRESHOLD = 0.01;
-    static constexpr double RESOLUTION_FACTOR       = 100.0;
+    static constexpr double CLOSE_TO_DEST_THRESHOLD =
+        0.01;  // if close to destination then return no path
+    static constexpr int BLOCKED_DESINATION_OSCILLATION_MITIGATION =
+        4;  // increase in threshold to reduce oscillation
+    static constexpr double BLOCKED_DESTINATION_SEARCH_RESOLUTION =
+        10.0;  // resolution for searching for unblocked point around a blocked
+               // destination
```
Comment (garethellis0) [inline src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.h]: Can we put these comments on the line above the variable? That way formatting won't mess this up

**Review by garethellis0 — DISMISSED** [review]
This looks good, just one comment. Can we confirm this works on the field before merging as well please?

File: src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.h (hunk 1 above)
Comment (jonathanlew) [reply src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.h]: ok

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Comment by jonathanlew** [comment]
Looks good on the field

**Pull request merged**